nixos/modules/services/torrent/qbittorrent.nix

```diff
@@ -0,0 +1,199 @@
+    };
+
+    group = mkOption {
+      type = str;
+      default = "qbittorrent";
+      description = "Group under which qbittorrent runs.";
+    };
+
+    profileDir = mkOption {
+      type = path;
+      default = "/var/lib/qBittorrent/";
+      description = "the path passed to qbittorrent via --profile.";
+    };
+
+    openFirewall = mkOption {
+      default = false;
+      description = "Opens both the webuiPort and torrentPort options ports as tcp in the firewall";
+    };
+
+    webuiPort = mkOption {
+      default = 8080;
+      type = port;
+      description = "the port passed to qbittorrent via `--webui-port`";
+    };
+
+    torrentingPort = mkOption {
+      type = nullOr port;
+      description = "the port passed to qbittorrent via `--torrenting-port`";
+    };
+
+    serverConfig = mkOption {
+      type = unspecified;
+      description = ''
+        Free-form settings mapped to the `qBittorrent.conf` file in the profile.
+        Refer to <https://github.com/qbittorrent/qBittorrent/wiki/Explanation-of-Options-in-qBittorrent>
+        you will probably want to run qBittorrent locally once to use the webui to generate the password before setting anything here.
+      '';
+      example = literalExpression ''
+        {
+          LegalNotice.Accepted = true;
+          Preferences = {
+            WebUI = {
+              Username = "user";
+              Password_PBKDF2 = "generated ByteArray.";
+            };
+            General.Locale = "en";
+          };
+        }
+      '';
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    systemd = {
+      tmpfiles.settings = {
+        qbittorrent = {
+          "${cfg.profileDir}/qBittorrent/"."d" = {
+            mode = "700";
+            inherit (cfg) user group;
+          };
+          "${cfg.profileDir}/qBittorrent/config/"."d" = {
+            mode = "700";
+            inherit (cfg) user group;
+          };
+          "${cfg.profileDir}/qBittorrent/config/qBittorrent.conf"."L+" = {
+            mode = "1500";
+            inherit (cfg) user group;
+            argument = "${pkgs.writeText "qBittorrent.conf" (gendeepINI cfg.serverConfig)}";
+          };
+        };
+      };
+      services.qbittorrent = {
+        description = "qbittorrent BitTorrent client";
+        wants = [ "network-online.target" ];
+        after = [
+          "local-fs.target"
+          "network-online.target"
+          "nss-lookup.target"
+        ];
+        wantedBy = [ "multi-user.target" ];
+
+        serviceConfig = {
+          Type = "simple";
+          User = cfg.user;
+          Group = cfg.group;
+          ExecStart = utils.escapeSystemdExecArgs ([
+              (getExe cfg.package)
+              "--profile=${cfg.profileDir}"
+              "--webui-port=${toString cfg.webuiPort}"
+            ]
+            ++ lib.optional (cfg.torrentingPort != null) "--torrenting-port=${toString cfg.torrentingPort}");
+          TimeoutStopSec = 1800;
```

Comment:
That is really really long and will delay shutdown. Can we decrease that to 60 seconds?

Reply:
The 30 min was intentional. And as per https://github.com/qbittorrent/qBittorrent/pull/15414#issuecomment-912270943, shutdown time was already considered. See also https://github.com/qbittorrent/qBittorrent/issues/15381#issuecomment-912080617.

Reply:
leaving as is for now then, thanks eclairevoyant